overlord/snapstate/autorefresh_gating_test.go

```diff
@@ -479,3 +479,30 @@ func (s *autorefreshGatingSuite) TestAffectedByBootBase(c *C) {
 				"core18": true,
 			}}})
 }
+
+func (s *autorefreshGatingSuite) TestCreateAutoRefreshGateHooks(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+
+	affected := map[string]*snapstate.AffectedSnapInfo{
+		"snap-a": {
+			Base:    true,
+			Restart: true,
+		},
+		"snap-b": {},
+	}
+
+	ts := snapstate.CreateGateAutoRefreshHooks(st, affected)
+	c.Assert(ts.Tasks(), HasLen, 2)
+	t := ts.Tasks()[0]
+	c.Assert(t.Kind(), Equals, "run-hook")
+	var data interface{}
+	c.Assert(t.Get("hook-context", &data), IsNil)
+	c.Check(data, DeepEquals, map[string]interface{}{"base": true, "restart": true})
```

Comment:
Perhaps it's useful to double check HookSetup too, that it's `gate-auto-refresh` and optional, and we saw one for each snap.

Reply:
+1